tests/unittests/Foundation/NSStringTests.m

```diff
@@ -607,4 +607,37 @@ void testUrlCharacterSetEncoding(NSString* decodedString, NSString* encodedStrin
 
     ASSERT_OBJCEQ(@"value1", propertyList[@"key1"]);
     ASSERT_OBJCEQ(@"value2", propertyList[@"key2"]);
+}
+
+TEST(NSString, StringByAddingPercentEscapesUsingEncodingTest) {
+    NSString* originalString = @"abcdefghijklmnopqrstuvwxyz : !@#$%^&*()_+1234567890-=`~\\|]}[{'\";:.>,</?";
+    NSString* expectedEscapedString =
+        @"abcdefghijklmnopqrstuvwxyz%20:%20!@%23$%25%5E&*()_+1234567890-=%60~%5C%7C%5D%7D%5B%7B'%22;:.%3E,%3C/?";
+
+    NSString* escapedString = [originalString stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
+
+    EXPECT_OBJCEQ(expectedEscapedString, escapedString);
+    EXPECT_OBJCEQ(originalString, escapedString.stringByRemovingPercentEncoding);
+    EXPECT_OBJCEQ(originalString, [escapedString stringByReplacingPercentEscapesUsingEncoding:NSUTF8StringEncoding]);
+
+    originalString = @"claesström";
+    expectedEscapedString = @"claesstr%C3%B6m";
+
+    escapedString = [originalString stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
+
+    EXPECT_OBJCEQ(expectedEscapedString, escapedString);
+    EXPECT_OBJCEQ(originalString, escapedString.stringByRemovingPercentEncoding);
+    EXPECT_OBJCEQ(originalString, [escapedString stringByReplacingPercentEscapesUsingEncoding:NSUTF8StringEncoding]);
+
+    // Removing percent encoding is unsupported on the reference platform for anything other than UTF8 which is why the reference platform
```

Comment:
>which is why [](start = 103, length = 12)

I do not think that we can say with certainty that that is _why_ they deprecated it.